Source/WebCore/page/WorkerNavigator.cpp

```diff
@@ -65,28 +65,15 @@ GPU* WorkerNavigator::gpu()
 {
 #if HAVE(WEBGPU_IMPLEMENTATION)
     if (!m_gpuForWebGPU) {
-        auto scriptExecutionContext = this->scriptExecutionContext();
-        if (scriptExecutionContext->isWorkerGlobalScope()) {
-            WorkerGlobalScope& workerGlobalScope = downcast<WorkerGlobalScope>(*scriptExecutionContext);
-            if (!workerGlobalScope.graphicsClient())
-                return nullptr;
-
-            RefPtr gpu = workerGlobalScope.graphicsClient()->createGPUForWebGPU();
-            if (!gpu)
-                return nullptr;
-
-            m_gpuForWebGPU = GPU::create(*gpu);
-        } else if (scriptExecutionContext->isDocument()) {
-            Ref document = downcast<Document>(*scriptExecutionContext);
-            RefPtr page = document->page();
-            if (!page)
-                return nullptr;
-            RefPtr gpu = page->chrome().createGPUForWebGPU();
-            if (!gpu)
-                return nullptr;
-
-            m_gpuForWebGPU = GPU::create(*gpu);
-        }
+        Ref context = downcast<WorkerGlobalScope>(*this->scriptExecutionContext());
```

Comment:
Does this work for DedicatedWorkers too? It looks like the:
```
} else if (scriptExecutionContext->isDocument()) {
            Ref document = downcast<Document>(*scriptExecutionContext);
```
was added for dedicated workers, but maybe it is not needed?

Reply:
Dedicated workers also use `WorkerGlobalScope`. They're not documents...

Reply:
Sorry I just checked, indeed none of the workers take the removed path.